components/script/dom/submitevent.rs

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::codegen::Bindings::EventBinding::EventMethods;
+use crate::dom::bindings::codegen::Bindings::SubmitEventBinding;
+use crate::dom::bindings::codegen::Bindings::SubmitEventBinding::SubmitEventMethods;
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::bindings::str::DOMString;
+use crate::dom::event::Event;
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::htmlelement::HTMLElement;
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+use servo_atoms::Atom;
+
+#[dom_struct]
+#[allow(non_snake_case)]
+pub struct SubmitEvent {
+    event: Event,
+    submitter: Option<DomRoot<HTMLElement>>,
+}
+
+impl SubmitEvent {
+    pub fn new_inherited(submitter: Option<DomRoot<HTMLElement>>) -> SubmitEvent {
+        SubmitEvent {
+            event: Event::new_inherited(),
+            submitter: submitter,
+        }
+    }
+
+    pub fn new_initialized(
```

Comment:
No need for a separate method here; let's inline this code into `new` instead.

Reply:
Right, that's me copy-and-pasting from other event code a little too heavily.